### ISSUE TITLE
CSS Parsing: Large delay when opening the visual editor when a lot of CSS is present

### ISSUE THREAD

**Issue by AlexGStapleton** [siteorigin/so-css#44]
When a lot of CSS (3109 lines to be exact) is present the delay to open the visual editor is much larger than for the old CSS parsing library.

Before:
![](https://i.imgur.com/UxglUlj.gif)

Now:
![](https://i.imgur.com/bHQRCfD.gif)

(please note that due to the encoding, these gifs aren't exactly reliable - you'll need to replicate this manually to truely understand).

People use Custom CSS as a child theme replacement so it's to be expected certain people will have very large amounts of CSS present.

Reference CSS (duplicated to hit slightly over 3k lines):
```
/* BLOCKQUOTE FORMATTING*/
blockquote {
  color: #000000;
  background: #eee;
  margin: 0 0 1.6em 0.8em;
  padding: 1.4em 1em 0.1em 1em;
  border-left: 5px solid #7F1315;
}

blockquote:before {
  display: none;
}

/* INCREASE SPACE BETWEEN PARAGRAPHS */
#content p {
  margin-bottom: 15px;
}

/*ALIGN VIEW CART LINK*/
a.added_to_cart.wc-forward {
  position: absolute;
  display: block;
  bottom: -30px;
  margin-left: 60px;
}

/*ALIGN ADD TO CART BUTTONS*/
.woocommerce ul.products li.product, .woocommerce-page ul.products li.product {
  min-height: 440px !important;
  margin-bottom: 70px;
}

a.button.add_to_cart_button, a.button.product_type_simple {
  bottom: 0px;
  position: absolute;
  margin-left: -65px;
}

/* BOLD SHOP PRICES */
.amount {
  font-size: 18px;
  font-weight: bold;
}
```

**Comment by braamgenis** [comment]
Issue only in #43 

@AlexGStapleton Cool, will check this out. I hope I can actually do something about it. :suspect:

**Comment by AlexGStapleton** [comment]
In all likelihood, the previous parser likely had a check in place to avoid reprocessing it (this explains the slight delay on first visual open), while the new one doesn't.

**Comment by braamgenis** [comment]
@AlexGStapleton I've made some changes which should make it faster to open. It's still slowish when closing, due to setting the large amount of CSS back on codemirror, but it's not simple to do anything about that at this point and it's a pre-existing issue. Please check and let me know.

**Comment by AlexGStapleton** [comment]
It definitely feels faster to open after initial open.

![](https://i.imgur.com/ZCFODtn.gif)

**Comment by braamgenis** [comment]
Hmm, the initial open should be faster too... Could you please make sure the old JS file isn't cached. Then if the initial open still feels slow, please send a copy of the CSS you're using. Thanks! :metal:

**Comment by AlexGStapleton** [comment]
Cleared cache - It's about the same. :(

The CSS I'm using is the reference CSS duplicated until 3k lines.

**Comment by braamgenis** [comment]
I'm really confused, that change should definitely have made at least some difference to the initial open. Taking a long shot here, but could you please open the `editor.js` file to [line 765](https://github.com/siteorigin/so-css/blob/develop/js/editor.js#L765) in your local copy and check that that line is there. I.e. it should be `var valueChanged = false;`

**Comment by AlexGStapleton** [comment]
I uh, may have deleted it. 🙍 

Regardless, I just downloaded the latest version of develop (previously I downloaded the relevant branch - but it's been merged and junk) and it looks like I had the old one as it's now super fast. 😖 Admitally, this was due to it removing all of the duplicated CSS but hey, it's duplicated CSS - who cares.

So I copied the entire contents of Vantage's style.css and it was much faster to load than before, but as you stated, closing it was still a tad slow. Still much better than before though.
